src/components/button/button.js

```diff
@@ -11,6 +11,8 @@
 import '../../vendor/polyfills/Event' // addEventListener and event.target normaliziation
 
 var KEY_SPACE = 32
+var DEBOUNCE_TIMEOUT_IN_SECONDS = 1
+var debounceFormSubmitTimer = null
```

Comment:
Going through this scenario out loud...

From the server:

```html
<form action="/delete/1/">
  <button type="submit">Delete 1</button>
</form>
```

When JavaScript enhances the page, I suppose you might turn this into a "button" type, but more likely you'd just prevent the default submission.

```html
<form action="/delete/1/">
  <button type="button">Delete 1</button>
</form>
```

```js
document.querySelector('form').addEventListener('submit', event => {
  // Do an AJAX request
  // ...
  event.preventDefault()
})
```

Reply:
So I suppose there's definitely a chance that, you could want to rattle through deleting lots of things...

Reply:
If we wanted to have multiple timers we could change how this module works from

```
new Button(document).init()
```

to (pseudo code)

```js
Array.from(document.querySelector('button, input=button, [role=button]')).forEach($button => {
  new Button($button).init()
})
```

Then the timer instance could be stored on the button instance itself.

Reply:
alternative markup to consider coming from the server, not sure if this would affect your thinking

```
<form action="/delete">
  Item 1 <button name="id" value="1" type="submit">Delete 1</button>
  Item 2 <button name="id" value="2" type="submit">Delete 2</button>
  Item 3 <button name="id" value="3" type="submit">Delete 3</button>
</form>
```

I think it would be reasonable to support deleting multiple items without waiting for a response between each button click for both JS and non-JS users.